sdk/data/azcosmos/cosmos_transactional_batch_response.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azcosmos
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+)
+
+// TransactionalBatchResponse contains response from a transactional batch operation.
+type TransactionalBatchResponse struct {
+	Response
+	// SessionToken contains the value from the session token header to be used on session consistency.
+	SessionToken string
+	// OperationResults contains the individual batch operation results.
+	// The order of the results is the same as the order of the operations in the batch.
+	OperationResults []TransactionalBatchResult
+	// Success indicates if the transaction was successfully committed.
+	// If false, one of the operations in the batch failed.
+	// Inspect the OperationResults, any operation with status code http.StatusFailedDependency is a dependency failure.
+	// The cause of the batch failure is the first operation with status code different from http.StatusFailedDependency.
+	Success bool
+}
+
+func newTransactionalBatchResponse(resp *http.Response) (TransactionalBatchResponse, error) {
+	response := TransactionalBatchResponse{
+		Response: newResponse(resp),
+	}
+
+	response.SessionToken = resp.Header.Get(cosmosHeaderSessionToken)
+
+	response.Success = resp.StatusCode != http.StatusMultiStatus
+
+	if err := runtime.UnmarshalAsJSON(resp, &response.OperationResults); err != nil {
+		return TransactionalBatchResponse{}, err
+	}
+
+	return response, nil
+}
+
+// TransactionalBatchResult represents the result of a single operation in a batch.
+type TransactionalBatchResult struct {
+	// StatusCode contains the status code of the operation.
+	StatusCode int32
```

Comment:
You can add tags to each field to handle JSON marshalling/unmarshalling which should allow you to remove your custom unmarshaller.  Here's an example.
```go
StatusCode int32 `json:"statusCode"`
```
See the [docs](https://pkg.go.dev/encoding/json#Marshal) for more info.

Reply:
I tried the tags. The reason I need the Unmarshall is that the body of a particular item we are returning it as a `[]byte` to be inline with all the other item responses (ItemResponse). And that needs to be obtained from the server response for that particular item. The server returns a JSON array, each element contains attributes (like the statusCode) and a JSON object as the body, I need to convert the JSON body into a []byte. The `json` decorator does not work on that scenario.